designsafe/apps/api/agave/views.py

```diff
@@ -89,7 +90,7 @@ def get(self, request, file_mgr_name, system_id=None, file_path=''):
                                 encoder=AgaveJSONEncoder,
                                 safe=False)
 
-
+@method_decorator(csrf_exempt, name='dispatch')
```

Comment:
this is exempting all of these methods as csrf_exempt -- you should apply it to only the public media views

Reply:
Good catch, thanks! I missed that in the docs. It says I need to put the csrf_exepmtion on the dispatch, but later it says you also need to put csrf_protect/exempt on all the protected/non protected methods like PUT POST etc.. I'll get this sorted out.

Reply:
Note: It may not work to exempt the CSRF token for downloads and previews, because this request uses a PUT method. Still working on this one...